### PR TITLE
MAINT: special: reimplement inverses of incomplete gamma functions

### PR DESCRIPTION
**Pull request by person142**
This primarily improves the accuracy of `gammaincinv` at small
arguments. (Previously the small argument test was a known failure.)
The implementation is essentially ported from Boost with some changes
in the implementation of Halley's method.

### CODE REVIEW

**Comment by person142** [comment]
Note that the adjustments needed for the doctest failure appear to be an issue with `refguide_check.py`, which seems to be falling back on string comparison instead of making numerical comparisons. (Probably because all the numbers are wrapped in a class it doesn't understand.)

**Comment by person142** [comment]
Bump. I know there aren't too many folks into reviewing `special` PRs, but maybe someone will feel inspired. ;-)

File: scipy/special/cephes/igami.c [hunk 1]
```diff
@@ -1,180 +1,342 @@
...
+             * the first form given here is unstable when p is close to 1,
+             * making it impossible to compute the inverse of Q(a,x) for small
+             * q. Fortunately the second form works perfectly well in this case.
+             */
+            double u;
+            if((b * q > 1e-8) && (q > 1e-5)) {
+                u = pow(p * g * a, 1 / a);
+            }
+            else {
+                u = exp((-q / a) - NPY_EULER);
```
Comment (pv) [inline scipy/special/cephes/igami.c]: No test coverage for this branch, according to `runtests.py --gcov -s special; runtests.py --lcov-html`

File: scipy/special/cephes/igami.c [hunk 2]
```diff
@@ -1,180 +1,342 @@
...
+            if (w < 3 * a) {
+                result = w;
+            }
+            else {
+                double D = fmax(2, a * (a - 1));
+                double lg = lgam(a);
+                double lb = log(q) + lg;
+                if (lb < -D * 2.3) {
+                    /* DiDonato and Morris Eq 25: */
+                    double y = -lb;
```
Comment (pv) [inline scipy/special/cephes/igami.c]: No test coverage for this branch.

**Review by pv — APPROVED** [review]
Ok, looks good to me. Some branches untested though, but checked the copypaste from boost is correct.

**Comment by person142** [comment]
Thanks for reviewing @pv.

**Pull request merged**